src/qrcode/QRCode.tsx

```diff
@@ -15,19 +15,23 @@ import { currentAccountSelector } from 'src/web3/selectors'
 
 interface Props {
   qrSvgRef: React.MutableRefObject<SVG>
+  dataType: QRCodeDataType
 }
 
-const mapStateToProps = (state: RootState): Partial<UriData> => ({
+const mapStateToProps = (state: RootState) => ({
   address: currentAccountSelector(state)!,
   displayName: nameSelector(state) || undefined,
   e164PhoneNumber: state.account.e164PhoneNumber || undefined,
 })
 
-export default function QRCodeDisplay({ qrSvgRef }: Props) {
+export default function QRCodeDisplay({ qrSvgRef, dataType }: Props) {
   const data = useSelector(mapStateToProps, shallowEqual)
+
+  // ValoraDeepLink generates a QR code that deeplinks into the walletconnect send flow of the valora app
+  // Address generates a QR code that has the walletAddress as plaintext that is readable by wallets such as Coinbase and Metamask
   const qrContent = useMemo(
-    () => urlFromUriData(data),
-    [data.address, data.displayName, data.e164PhoneNumber]
+    () => (dataType === QRCodeDataType.ValoraDeepLink ? urlFromUriData(data) : data.address),
```

Comment:
could be nice if this was a switch that threw an error if the dataType was not recognized, so it wouldn't just fail silently if the dataType was some other QRCodeDataType that we forgot to account for here.

could even get a compile-time error if there was some object mapping QRCodeDataType keys to functions that get qr contents from `data`

Reply:
you could also get a compile-time error if there was an object mapping QRCodeDataType keys to expected QR code contents in QRCode.test.tsx, which could be nicer